src/Board.java

```diff
@@ -97,4 +106,40 @@ private void updatePieceCount(int position) {
             this.numberOfBlackPieces--;
         }
     }
+    
+    public boolean isEndState() {
+        return (numberOfBlackPieces == 0 || numberOfWhitePieces == 0);
+    }
+
+    public List<Square> getAdjacentSquares(Square square) {
+        List<Integer> squareNumbers = square.getAdjacentSquares();
+        return getSquares((ArrayList<Integer>) squareNumbers);
+    }
+
+    public List<Square> getSquaresTwoMovesAway(Square square) {
```

Comment:
Beautiful. That'll be very helpful later when we have to come back and read this code again to fix the bugs and logic we inevitably will miss. :)